Deprecation of Mambaforge

It seems like mamabforge is deprecated and is starting to "encourage" users to switch to miniforge with brownouts ([see here](https://conda-forge.org/news/2024/07/29/sunsetting-mambaforge/)).

Somewhere in the CI it seems like mambaforge is used which causes the jobs to fail (see  https://github.com/openmm/openmm/actions/runs/11126960053 or https://github.com/openmm/openmm/actions/runs/11116574621/job/30887087669 ).

We should switch to miniforge asap if we want the CI to work in the future. I'm not familiar with github actions/CI so I'm probably not a big help here though.

I changed it in #4678.  Unfortunately, it's still failing:

```
2024-10-01T15:41:31.1066416Z Will fetch Miniforge3 Miniforge3 from https://github.com/conda-forge/miniforge/releases/download/Miniforge3/Miniforge3-Miniforge3-Linux-x86_64.sh
2024-10-01T15:41:31.1067780Z Ensuring Installer...
2024-10-01T15:41:31.1070255Z Checking for cached Miniforge3@Miniforge3...
2024-10-01T15:41:31.1076927Z Did not find Miniforge3-Miniforge3-Linux-x86_64.sh Miniforge3 in cache
2024-10-01T15:41:31.4038126Z ##[endgroup]
2024-10-01T15:41:31.4065117Z ##[error]Unexpected HTTP response: 404
```

Perhaps it's an unrelated error?  Or maybe they messed up their block, and it's preventing miniforge downloads as well as mambaforge?  Or did I do something wrong?

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -191,7 +191,7 @@ jobs:
           activate-environment: build
           environment-file: devtools/ci/gh-actions/conda-envs/build-${{ matrix.env }}.yml
           auto-activate-base: false
-          miniforge-version: Miniforge-pypy3
+          miniforge-variant: Miniforge-pypy3
           use-mamba: true
 
       - uses: conda-incubator/setup-miniconda@v3
@@ -202,7 +202,7 @@ jobs:
           activate-environment: build
           environment-file: devtools/ci/gh-actions/conda-envs/build-${{ matrix.env }}.yml
           auto-activate-base: false
-          miniforge-version: Miniforge3
+          miniforge-variant: Miniforge3
           use-mamba: true
 
       - name: "Install CUDA on Ubuntu (if needed)"

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -191,7 +191,7 @@ jobs:
           activate-environment: build
           environment-file: devtools/ci/gh-actions/conda-envs/build-${{ matrix.env }}.yml
           auto-activate-base: false
-          miniforge-variant: Mambaforge-pypy3
+          miniforge-version: Miniforge-pypy3
           use-mamba: true
 
       - uses: conda-incubator/setup-miniconda@v2
@@ -202,7 +202,7 @@ jobs:
           activate-environment: build
           environment-file: devtools/ci/gh-actions/conda-envs/build-${{ matrix.env }}.yml
           auto-activate-base: false
-          miniforge-variant: Mambaforge
+          miniforge-version: Miniforge3
           use-mamba: true
 
       - name: "Install CUDA on Ubuntu (if needed)"

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -184,7 +184,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - uses: conda-incubator/setup-miniconda@v2
+      - uses: conda-incubator/setup-miniconda@v3
         name: "Prepare base dependencies"
         if: matrix.python-version == 'pypy'
         with:
@@ -194,7 +194,7 @@ jobs:
           miniforge-version: Miniforge-pypy3
           use-mamba: true
 
-      - uses: conda-incubator/setup-miniconda@v2
+      - uses: conda-incubator/setup-miniconda@v3
         name: "Prepare base dependencies"
         if: matrix.python-version != 'pypy'
         with:

